elements/powerprediction.lua

```diff
@@ -0,0 +1,161 @@
+--[[ Element: Power Prediction Bar
+ Handles updating and visibility of the power prediction status bars.
+
+ Widget
+
+ PowerPrediction - A table containing `mainBar` and `altBar`.
+
+ Sub-Widgets
+
+ mainBar - A StatusBar used to represent power cost of spells, that consume
+           your main power, e.g. mana for mages;
+ altBar  - A StatusBar used to represent power cost of spells, that consume
+           your additional power, e.g. mana for balance druids.
+
+ Notes
+
+ The default StatusBar texture will be applied if the UI widget doesn't have a
+ status bar texture.
+
+ Examples
+
+   -- Position and size
+   local mainBar = CreateFrame('StatusBar', nil, self.Power)
+   mainBar:SetReverseFill(true)
+   mainBar:SetPoint('TOP')
+   mainBar:SetPoint('BOTTOM')
+   mainBar:SetPoint('RIGHT', self.Power:GetStatusBarTexture(), 'RIGHT')
+   mainBar:SetWidth(200)
+
+   local altBar = CreateFrame('StatusBar', nil, self.DruidMana)
```

Comment:
It seems right how Val did it. E.g. shamans have maelstrom as their primary resource, which is displayed on the standard powerbar, and mana as a secondary resource for heals, which oUF would display as the DruidMana element. Maelstrom spells would drain the main powerbar (self.Power) and heals would drain the secondary one (self.DruidMana).